doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,619 @@
+  If no matches can be found in the path, the ``EXPRESSION_IF_FALSE`` branch is applied.
+  Identifiers in the query expression are not applied, e.g. if a query expression queries for a specific policy, the specific policy
+  is not applied to the hops it matches, unless the same expression is also given under the ``EXPRESSION_IF_TRUE`` branch.
+  An identifier evaluates to true when at least a single hop in the path matches the identifier and a concatenation evaluates to true when
+  *all* identifiers in the concatenation also evaluate to true.
+
+  To illustrate this, see the ``Example`` using the path ``1-ff00:0:109#0,5@() 1-ff00:0:110#4,1@(G100, G200) -> 1-ff00:0:111#2,0@(G200, G300)``:
+
+  When an expression queries for ``1-0#0,0@G200`` using ``{1-0#0,0@G200 ? 1-0#0,0@G300 : 1-0#0,0@REJECT}``, the policies that are
+  applied to the hops are only policy G300 for the last hop.
+  To also apply policy G200, the query has to be structured as ``{1-0#0,0@G200 ? (1-0#0,0@G300 + 1-0#0,0@G200) : 1-0#0,0@REJECT}``.
+
+  When a query is used within another query, the query_expression is first used to determine which branch is used for the result.
+  If the query would apply the ``EXPRESSION_IF_TRUE`` branch, the result of the query is the evaluation of the ``EXPRESSION_IF_TRUE`` branch.
+  The same applies for the ``EXPRESSION_IF_FALSE`` branch.
+
+**Evaluation Order**
+The language is evaluated left to right, for each hop only a single policy can be applied.
+The first identifier match applies the policy, so the order of the query is important. 
+
+Example:
+
+``(0-0#0,0@REJECT + 1-0#0,0@00)``
+
+Will reject all paths, whereas
+
+``(1-0#0,0@0 + 0-0#0,0@REJECT)``
+
+Will reject all paths that are not within ISD 1.
+
+Once the application has decided which policies to use, it can craft a FABRID HBH extension and include this as an option when sending
+the packet.
+
+DRKey
+^^^^^^
+
+FABRID uses DRKey for computing the Encrypted Policy Indices, the FABRID Hop Validation Fields and the Path Validator.
+The routers use the fast key derivation side, whereas the endhosts will use the slow side.
+
+Configuration
+--------------
+
+Control service
+^^^^^^^^^^^^^^^^^^
+
+To be able to use DRKey, one has to configure the control service setting *drkey.level1_db* and *drkey.secret_value_db*.
+Additionally, since the border routers will fetch the secret value from the control service, the control service also has to
+add the internal IP address of all border routers of the local AS to the DRKey delegation list for FABRID.
+
+Example (cs1-ff00_0_110-1.toml)::
+
+    [drkey.level1_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-level1.db"
+
+    [drkey.secret_value_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-secret.db"
+
+    [drkey.delegation]
+    FABRID = [ "fd00:f00d:cafe::7f00:11", "fd00:f00d:cafe::7f00:12", "fd00:f00d:cafe::7f00:13"]
+
```

Comment:
```suggestion

The control service has to know the folder in which it can find the FABRID policy configurations. This is configured using *fabrid_path*.

Example (cs1-ff00_0_110-1.toml)::

    [general]
    fabrid_path = "gen/ASff00_0_110/fabrid/"
    
```